flang/test/Lower/OpenMP/generic-loop-rewriting.f90

```diff
@@ -0,0 +1,45 @@
+!RUN: %flang_fc1 -emit-hlfir -fopenmp %s -o - | FileCheck %s
```

Comment:
When this is ready for review, it would be good to see an mlir-to-mlir test which runs only the new pass. This makes it easier to review exactly what the pass is doing. It is up to you if you keep this lowering test as well.